Merge Sort in Python

ranesr opened the pull request:
This pull request is for issue number #33

ranesr committed:
Merge Sort in Python

ranesr committed:
Improving readability for MergeSort.py

VamsiSangam commented on Sorting Algorithms/Merge Sort Algorithm/Python/MergeSort.py:
```diff
@@ -0,0 +1,57 @@
+		j = 0
+		k = 0
+		
+		# Merge the two halves
```
Could put the merge part of the code in a separate function ```merge``` and then put ``` merge(array, left_half, right_half) ``` here?

ranesr committed:
Updating MergeSort.py

Adding separate merge function as suggested.

VamsiSangam commented on Sorting Algorithms/Merge Sort Algorithm/Python/MergeSort.py:
```diff
@@ -25,33 +25,31 @@ def merge_sort(array):
+		return left
+	
+	# Compare first element and merge the remaining elements of the part (with smaller element) with other part
+	if left[0] < right[0]
```
You are missing a ```:``` here. Please correct.

ranesr committed:
Fixed code

VamsiSangam commented on Sorting Algorithms/Merge Sort Algorithm/Python/MergeSort.py:
```diff
@@ -44,7 +45,7 @@ def merge(left, right):
+	if left[0] < right[0]:
 		return left[0] + merge(left[1:], right)
 		
 	return right[0] + merge(left, right[1:])
```
You are returning an ```int``` here and a ```list``` in line 45, so this line gives an error saying ``` TypeError: unsupported operand type(s) for +: 'int' and 'list' ``` ... And please implement the merge logic using loops as you did earlier. This seems fine but calls ```merge``` too many times for large input.

ranesr committed:
Updated MergeSort

Updated original merge function and added another merge function using loops. Added example for both merge function.

VamsiSangam commented:
Thanks a lot! Your code should be live on the [corresponding post](http://theoryofprogramming.com/2015/01/29/merge-sort-algorithm/) ! :smile:
